app/lib/exchange-rate/money.ts

```diff
@@ -0,0 +1,458 @@
+import { Big } from 'big.js';
+
+export type NumberInput = number | string | Big;
+
+export type MoneyInput = {
+  /**
+   * Money amount
+   */
+  amount: NumberInput;
+  /**
+   * Currency for the provided amount of money
+   */
+  currency: string;
+  /**
+   * Unit of currency to use. For example for USD it can be 'dollar' or 'cent', for BTC 'bitcoin', 'satoshi' or
+   * 'millisatoshi', etc.
+   * If not provided the default/base unit is used (bitcoin for BTC, dollar for USD, etc.)
+   */
+  unit?: string;
+};
+
+type FormatOptions = {
+  locale?: string;
+  currency?: string;
+};
+
+type CurrencyUnit = {
+  name: string;
+  decimals: number;
+  symbol: string;
+  factor: Big;
+  format: (value: number, options?: FormatOptions) => string;
+};
+
+type CurrencyData = {
+  baseUnit: string;
+  units: CurrencyUnit[];
+};
+
+type BaseFormatOptions = FormatOptions & { decimals: number };
+
+function baseFormat(value: number, options: BaseFormatOptions) {
+  const { locale, decimals, currency } = options;
+  const formatOptions: Parameters<typeof Intl.NumberFormat>[1] = {
+    minimumFractionDigits: decimals,
+    maximumFractionDigits: decimals,
+  };
+  if (currency) {
+    formatOptions.style = 'currency';
+    formatOptions.currency = currency;
+  }
+  return Intl.NumberFormat(locale, formatOptions).format(value);
+}
+
+const currencyDataMap: Record<string, CurrencyData> = {
+  USD: {
+    baseUnit: 'dollar',
+    units: [
+      {
+        name: 'dollar',
+        decimals: 2,
+        symbol: '$',
+        factor: new Big(1),
+        format: function (value: number, options: FormatOptions = {}) {
+          return baseFormat(value, { ...options, decimals: this.decimals });
+        },
+      },
+      {
+        name: 'cent',
+        decimals: 0,
+        symbol: '¢',
```

Comment:
I am not sure if this is the symbol we want and if we also want to put it after the number but that's what I did for now. Should be easy to change

Reply:
ok, keeping it then